Replace js-cookie

dunxd opened the pull request:
Remove requirement for js-cookie by rewriting cookie functions in util.js.  Also removes links to js-cookie from template files and also removes js-cookie from composer.json as it is no longer needed anywhere.

dunxd commented:
SonarCloud check is suggesting for-of loop and replacing `var` with `let` etc. but these aren't supported in EC5 browsers.  Is there a way to tell SonarCloud that util.js needs to work in EC5 browsers?

mikespub commented:
You need to be very careful about what ChatGPT/Copilot generates and double-check everything, as it often misses obvious things - for example, the fact that js-cookie functions are called directly in the templates too. Try a simply search for "Cookies." in the code base and you'll see it...

About SonarCloud, I'm still experimenting with it but I haven't found an option to let it be compatible with ES5 instead of ES6+ without disabling a bunch of Javascript rules. Either way, I'll evaluate the results on a case-by-case basis -  given the amount of legacy code in here, I don't intend to use its Quality Gate as some kind of automatic gatekeeper :-)

mikespub commented:
I've excluded util.js for future checks

mikespub commented:
Replace with legacy browser cookies - see #140

dunxd commented:
I've removed the remaining references to js-cookies - they were the button to set to the default template in the footers.  I also added a default "days" to the setCookie so it doesn't always need to be specified.  

Note that updateCookie() in util.js is very specific to use on the customize page - it always was but now it may be misunderstood by future developers since it sites next to the very generic setCookie() and eraseCookie() functions.  Maybe it would be clearer if renamed to updateCookieOnFormChange().

dunxd commented:
Hmm - not working properly now.  I don't know if it is worth putting the effort in to fix this since the alternative method that delivers legacy js-cookie to browser identifying itself in User Agent string as Kindle is working in the main branch. Maybe that is more maintainable in the long run.

mikespub commented:
hi @dunxd that was my thinking too...

mikespub commented:
Alternative from main branch
